crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs

```diff
@@ -765,6 +765,29 @@ pub fn update_top_call_execution_resources(runtime: &mut ForgeRuntime) {
     top_call.borrow_mut().used_execution_resources = all_execution_resources;
 }
 
+pub fn update_top_call_onchain_data(runtime: &mut ForgeRuntime) {
```

Comment:
This is done below in `get_all_used_resources` in a different manner
Why is this method needed?

Reply:
It is not the same - in this function we add messages to the top call in the trace. In the one below we obtain payloads of all messages from all calls (mind the `inner_calls: starknet_runtime.hint_handler.inner_calls`) via blockifier method and add them to cumulative resources